### PR TITLE
Windows non-ascii path fix

### PR DESCRIPTION
**Pull request by Zyx-2000**
I made it so VCMI can handle windows usernames with non-ascii characters.

### CODE REVIEW

**Comment by alexvins** [comment]
I don`t like this solution. Short path names are kind of deprecated and may not works on some windows 8.1+ installations.

**Comment by Zyx-2000** [comment]
I didn't know that. Perhaps I should make it so it only uses short paths if it detects a non-ascii character in the path. The cases where that works, will be a strict superset of the cases where the current code works.

**Comment by alexvins** [comment]
The proper for this problem should be fixing unicode support in VCMI VFS.

**Comment by alexvins** [comment]
Related bugs:
http://bugs.vcmi.eu/view.php?id=1688
http://bugs.vcmi.eu/view.php?id=1288

File: AI/BattleAI/BattleAI.cbp [hunk 1]
```diff
@@ -49,6 +49,7 @@
 		</Build>
 		<Compiler>
 			<Add option="-pedantic" />
+			<Add option="-std=c++11" />
```
Comment (alexvins) [inline AI/BattleAI/BattleAI.cbp]: wrong option, should be -std=gnu++11 for gcc

**Pull request closed without merging**